### PR TITLE
fix build instructions for debian:stable

### PR DESCRIPTION
**Pull request by rdanitz**
… and fix Dockerfile to build against HEAD

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,14 +13,14 @@ set(OAUTH2_TEST_CLIENT_SECRET "" CACHE STRING "Client Secret for testing Auth+ a
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake-modules")
 
 # clang-check and clang-format
-find_program(CLANG_FORMAT NAMES clang-format clang-format-3.7 clang-format-3.6)
+find_program(CLANG_FORMAT NAMES clang-format clang-format-3.7 clang-format-3.6 clang-format-3.5)
```
Comment (cajun-rat) [inline CMakeLists.txt]: Will clang-format-3.5 actually work?  I seem to remember that old versions of clang-format couldn't read the config file we are using and fail.  clang-format isn't needed for the build (it is only used for the development 'make qa') and the build should work fine without it. Does it fail for you?

File: Dockerfile [hunk 2]
```diff
@@ -17,7 +18,7 @@ RUN apt-get update -q && apt-get -qy install \
 
 COPY . /src/
 WORKDIR /src/
-RUN cmake .
+RUN cmake -DCMAKE_CXX_FLAGS="-std=c++11"
```
Comment (cajun-rat) [inline Dockerfile]: Good catch, although I would prefer for add a line like 'set_property(TARGET bla PROPERTY CXX_STANDARD 11)' to CMakeLists.txt. Grab me and we can make the change together.

File: CMakeLists.txt (hunk 1 above)
Comment (rdanitz) [inline CMakeLists.txt]: ahh yes, you're right, I only get a warning, but the actual build issue was caused by one of the other things

File: Dockerfile (hunk 2 above)
Comment (rdanitz) [inline Dockerfile]: cmake in the the debian:stable repo that we're using in the docker build has version 3.0.x whereas unfortunately this directive you mention is only supported in cmake >v3.1.

I pushed an update; this is the best I could come up with.

**Comment by cajun-rat** [comment]
Review score: 11/11 

Nice!

**Pull request merged**